thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/detector/email/filter/BaseAlertFilter.java

```diff
@@ -50,7 +64,12 @@ public void setParameters(Map<String, String> parameterSetting) {
         Field field = c.getDeclaredField(fieldName);
         boolean accessible = field.isAccessible();
         field.setAccessible(true);
-        field.set(this, Double.valueOf(value));
+        if (field.getType().equals(Double.class) || field.getType().equals(double.class)) {
+          field.set(this, value);
+        }
+        else {
```

Comment:
Maybe check whether it's String-compatible here - and throw an Exception if not.